src/core/server/saved_objects/migrations/core/index_migrator.ts

```diff
@@ -45,23 +46,66 @@ export class IndexMigrator {
    * @returns {Promise<MigrationResult>}
    */
   public async migrate(): Promise<MigrationResult> {
-    const context = await migrationContext(this.opts);
+    const { dryRun, index, callCluster, log, batchSize } = this.opts;
+    if (dryRun) {
+      const indexInfo = await fetchInfo(callCluster, index);
+      if (indexInfo.exists === false) {
+        return {
+          status: 'skipped',
+          alias: this.opts.index,
```

Comment:
Would it be possible to extract this whole `if dryRun` block into a private method to increase readability or is it too complicated because of the possible return statement inside?